examples/booleansearchparser.py

```diff
@@ -0,0 +1,486 @@
+
+CONTRIBUTORS:
+- Steven Mooij
+- Rudolph Froger
+- Paul McGuire
+
+TODO:
+- add more docs
+- ask someone to check my English texts
+- add more kinds of wildcards ('*' at the beginning and '*' inside a word)?
+
+"""
+from pyparsing import Word, alphanums, Keyword, Group, Combine, Forward, Suppress, Optional, OneOrMore, oneOf
+from sets import Set
+import re
+import string
+alphabet_ranges = [
+    ##CYRILIC: https://en.wikipedia.org/wiki/Cyrillic_(Unicode_block)
+    [int("0400",16), int("04FF",16)],
+    ##THAI: https://en.wikipedia.org/wiki/Thai_(Unicode_block)
+    [int("0E00",16), int("0E7F",16)],
+    ##ARABIC: https://en.wikipedia.org/wiki/Arabic_(Unicode_block) (Arabic (0600–06FF)+ Syriac (0700–074F)+ Arabic Supplement (0750–077F) )
+    [int("0600",16), int("07FF",16)],
+    ##CHINESE: https://en.wikipedia.org/wiki/CJK_Unified_Ideographs_(Unicode_block)
+    [int("0400",16), int("09FF",16)],
+    #JAPANESE : https://en.wikipedia.org/wiki/Japanese_writing_system
+    [int("3040",16), int("30FF",16)],
+    #KOREAN : https://en.wikipedia.org/wiki/Hangul
+    [int("AC00",16), int("D7AF",16)],
+    [int("1100",16), int("11FF",16)],
+    [int("3130",16), int("318F",16)],
+    [int("3200",16), int("32FF",16)],
+    [int("A960",16), int("A97F",16)],
+    [int("D7B0",16), int("D7FF",16)],
+    [int("FF00",16), int("FFEF",16)],
+]
+class BooleanSearchParser:
+
+    def __init__(self,only_parse=False):
+        self._methods = {
+            'and': self.evaluateAnd,
+            'or': self.evaluateOr,
+            'not': self.evaluateNot,
+            'parenthesis': self.evaluateParenthesis,
+            'quotes': self.evaluateQuotes,
+            'word': self.evaluateWord,
+            'wordwildcardprefix': self.evaluateWordWildcardPrefix,
+            'wordwildcardsufix': self.evaluateWordWildcardSufix,
+        }
+        self._parser    = self.parser()
+        self.text       = ''
+        self.words      = []
+    
+    def parser(self):
+        """
+        This function returns a parser.
+        The grammar should be like most full text search engines (Google, Tsearch, Lucene).
+        
+        Grammar:
+        - a query consists of alphanumeric words, with an optional '*' 
+          wildcard at the end or the begining of a word
+        - a sequence of words between quotes is a literal string
+        - words can be used together by using operators ('and' or 'or')
+        - words with operators can be grouped with parenthesis
+        - a word or group of words can be preceded by a 'not' operator
+        - the 'and' operator precedes an 'or' operator
+        - if an operator is missing, use an 'and' operator
+        """
+        operatorOr = Forward()
+        
+        alphabet  = ( 
+            u'*'+
+            alphanums
+        )
+        #suport for non-wester alphabets
+        for r in alphabet_ranges:
+            alphabet += u''.join(unichr(c) for c in xrange(*r) if not unichr(c).isspace())
```

Comment:
I saw this PR and thought I should offer: I tried something similar to this and found that it was easier to define my alphabet using a Regex expression which _excludes_ special characters. I did this instead of using pyparsing.Word so that I could use a Regex with `+` -- in my tests this was more efficient than any of the other approaches I tried.

```
# ESCAPE = \ , ESCAPED_SPECIAL = ESCAPE + Word(OPERATOR | ESCAPE, exact=1)
# not included here for brevity
ATOM = Combine(
    OneOrMore(
        ESCAPED_SPECIAL |
        Regex(r"""[^'":\\()/+\-=&|><!{}\[\]^~? ]+""") |
        ESCAPE
    )
)
```